src/Events/Services/Interfaces/IInboundService.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+using Altinn.Platform.Events.Models;
+
+namespace Altinn.Platform.Events.Services.Interfaces
+{
+    /// <summary>
+    /// Interface to talk to the events service
+    /// </summary>
+    public interface IInboundService
+    {
+        /// <summary>
+        /// Save cloud event to persistent storage.
+        /// </summary>
+        /// <param name="cloudEvent">The cloudEvent to be saved</param>
+        /// <returns>Id for the created document</returns>
+        Task<string> Save(CloudEvent cloudEvent);
+
+        /// <summary>
+        /// Push cloud event to inbound queue.
+        /// Should only be attempted after saving to persistent storage. 
+        /// </summary>
+        /// <param name="cloudEvent">The cloudEvent to be queued</param>
+        /// <returns>Id for the queued event</returns>
+        Task<string> PostInbound(CloudEvent cloudEvent);
+
+        /// <summary>
+        /// Save a cloud event to persistent storage and
+        /// pushes to Inbound events queue in one operation. 
+        /// </summary>
+        /// <param name="cloudEvent">The cloudEvent to be stored</param>
+        /// <returns>Id for the cloudEvent in persistent storage</returns>
+        Task<string> SaveAndPostInbound(CloudEvent cloudEvent);
+
+        /// <summary>
+        /// Gets list of cloud events based on query params
+        /// </summary>
+        Task<List<CloudEvent>> GetAppEvents(string after, DateTime? from, DateTime? to, int partyId, List<string> source, List<string> type, string unit, string person, int size = 50);
```

Comment:
Not sure I like the moving of this function. Both this and the save function might belong in the deleted appEvents service interface for now. Only methods related to the inbound processing of instances should be defined in this interface ace.

Reply:
I don't really think of it as an inbox and outbox. That is in my mind a concept related to messages, which in addition to data have a sender and a receiver.

> An [event](https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md#event) includes context and data about an [occurrence](https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md#occurrence). Each occurrence is uniquely identified by the data of the event.
>Events represent facts and therefore do not include a destination, whereas messages convey intent, transporting data from a source to a given destination.
>https://github.com/cloudevents/spec/blob/main/cloudevents/primer.md

Reply:
With all these queues and operations to handle them we're breaking up the traditional layered application we're used to working with. One trick could be to do a virtual microservice split. Which applications would we get and what would each of them be responsible for? What would we change if we decided to make events into two applications?

I personally feel like the logic of handling everything related to sending events to subscribers could be a separate application. I my head this respinsibility starts when something appears in the events-inbound queue, or when the application is made aware of a new event. 

The other stuff related to new incomming events, having them stored, retrieved, listed/filtered could be an other application.

Reply:
I think of inbound as the start of the subscription capabilities of our system. Hypothetically, if we had zero subscribers it could make sense to pause/stop the inbound queue, since there is no need to send information about events to subscribers.

So while sending an event to subscibers definitively can be viewed as our event system's "outbox", I don't think that creating an event is the "inbox". That is just registering an event, in my head. The event don't become a message with a destination before we have subscribers, and that is something that happens after registering and creating an event.